benchmarks/benchmarks-core/src/jmh/java/io/micrometer/benchmark/core/ObservationBenchmark.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2022 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.benchmark.core;
+
+import java.util.concurrent.TimeUnit;
+
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.observation.DefaultMeterObservationHandler;
+import io.micrometer.core.instrument.observation.ObservationOrTimerCompatibleInstrumentation;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationRegistry;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.profile.GCProfiler;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@Fork(1)
+@Threads(4)
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.SampleTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+public class ObservationBenchmark {
+
+    SimpleMeterRegistry meterRegistry;
+
+    ObservationRegistry observationRegistry;
+
+    ObservationRegistry noopRegistry;
+
+    Timer timer;
+
+    @Setup
+    public void setup() {
+        this.meterRegistry = new SimpleMeterRegistry();
+        this.timer = Timer.builder("cached.timer").tag("abc", "123").tag("index", "0").register(meterRegistry);
```

Comment:
It looks like the cached timer has one more tag (`index`) compared to the non-cached timers. Was this intentional? Would it be better to compare with the same number of tags in all cases unless we are specifically trying to measure a difference due to more tags?

Reply:
That is a leftover from an experiment with JIT and `NoopObservation`, my mistake, hanks for catching it, removed.